### ISSUE TITLE
StencilJS - Typescript: Explicit type definition of enum is required

### ISSUE THREAD

**Issue by psamusev** [ionic-team/stencil#2243]
**Stencil version:**

```
 @stencil/core@1.8.11
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**

```
[ ERROR ]  TypeScript: src/components.d.ts:32:24
           Cannot find name 'BreadcrumbStyle'.

     L31:  'breadcrumbItems': Array<BreadcrumbItem>;
     L32:  'breadcrumbStyle': BreadcrumbStyle;
     L33:  'i18n': BreadcrumbI18n;

[37:07.4]  build failed in 5.07 s
```

**Expected behavior:**

```
[33:42.9]  build finished in 5.44 s
```

**Steps to reproduce:**

1) Generate pure stencil library project
2) Create a component with the code snippet from the `Related code` section
3) Run `npm run build`

**Related code:**

**breadcrumb.tsx**

```tsx
import { Component, Host, h, Prop } from '@stencil/core';
import {
  BreadcrumbStyle
} from './breadcrumb.data';

@Component({
  tag: 'wk-breadcrumb',
  styleUrl: 'breadcrumb.css,
  shadow: true
})
export class Breadcrumb {

  @Prop( {reflect: true})
  breadcrumbStyle = BreadcrumbStyle.Default;

  render() {
    return (
      <Host>
        <slot></slot>
      </Host>
    );
  }

}

```

**breadcrumb.data.ts**

```ts
export enum BreadcrumbStyle {
  Default = 'default',
  Inverse = 'inverse'
}
```

**Other information:**

It works if I define a property with an explicit type like 
```
  @Prop( {reflect: true})
  breadcrumbStyle: BreadcrumbStyle = BreadcrumbStyle.Default;
```
but since Typescript allows you to define type by setting default value, compiler also should handle this case

**Comment by glenveegee** [comment]
I'm getting the same thing particularly in the latest stencil version 1.12.2 following the recent upgrade.

**Comment by alaboudi** [comment]
Ya I started using stencil and the first component I wrote failed because enums were not supported :(

**Comment by CarlosBalladares** [comment]
Seems like you need to declare the type in the component class scope.
Even custom types declared outside fail.
My workaround for a typography component

```typescript
import { Component, h, Prop } from '@stencil/core';

enum Variant_ENUM {
  H1 = 'h1',
  H2 = 'h2',
  H3 = 'h3',
  H4 = 'h4',
  H5 = 'h5',
  H6 = 'h6',
  P = 'p',
  LINK = 'link',
}

@Component({ tag: 'remy-typography', styleUrl: 'typography.css', shadow: true })
export class TodoList {
  @Prop() variant: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'link';

  render() {
    return (
      <span
        class={{
          [Variant_ENUM.H1]: this.variant == Variant_ENUM.H1,
          [Variant_ENUM.H2]: this.variant == Variant_ENUM.H2,
          [Variant_ENUM.H3]: this.variant == Variant_ENUM.H3,
          [Variant_ENUM.H4]: this.variant == Variant_ENUM.H4,
          [Variant_ENUM.H5]: this.variant == Variant_ENUM.H5,
          [Variant_ENUM.H6]: this.variant == Variant_ENUM.H6,
          [Variant_ENUM.P]: this.variant == Variant_ENUM.P,
          [Variant_ENUM.LINK]: this.variant == Variant_ENUM.LINK,
        }}
      >
        <slot />
      </span>
    );
  }
}

```

**Comment by rwaskiewicz** [comment]
Relates https://github.com/ionic-team/stencil/issues/2865

**Comment by rwaskiewicz** [comment]
The fix for this issue has been released as a part of today's [v4.2.0 release](https://github.com/ionic-team/stencil/releases/tag/v4.2.0).
